### ISSUE TITLE
eplus: the logic for ending live streaming after timeout has been overridden by Streamlink since version 6.0.0 and everything is still working

### ISSUE THREAD

**Issue by pzhlkj6612** [pmrowla/streamlink-plugins#19]
TL;DR: we don't have to do anything. Streamlink will not be affected by our code.

---

Please read #12 first.

<br>

I was using Streamlink 6.1.0 with our plugin (f45a8a722fd088ec5c1b7cc830ce88ffafc2e3fe) and found a warning I haven't seen:

```
[00:54:55.18]Reloading playlist
[00:54:55.25]Adding segment 1 to queue
[00:54:55.29]Writing segment 1 to output
[00:54:55.29]Segment 1 complete
[00:55:01.18]Reloading playlist
[00:55:01.20]Adding segment 2 to queue
[00:55:01.22]Writing segment 2 to output
[00:55:01.22]Segment 2 complete
[00:55:07.18]Reloading playlist
[00:55:07.23]Adding segment 3 to queue
[00:55:07.27]Writing segment 3 to output
[00:55:07.27]Segment 3 complete
[00:55:13.18]Reloading playlist
[00:55:16.18]Reloading playlist
[00:55:19.18]Reloading playlist
[00:55:22.18]Reloading playlist
[00:55:25.18]Reloading playlist
[00:55:25.27][warning] No new segments in playlist for more than 18.00s. Stopping...
[00:55:25.27]Closing worker thread
[00:55:25.27]Closing writer thread
[00:55:25.27]Closing session updater...
[00:55:25.28][cli][info] Stream ended
[00:55:25.30][cli][info] Closing currently open stream...

```

Please note the "`No new segments in playlist for more than 18.00s. Stopping...`" warning. Our "`The 30.0-second timeout reached, this is the last playlist`" message was not shown.

That warning is introduced in Streamlink 6.0.0; the code I was using is:

https://github.com/streamlink/streamlink/blob/6.1.0/src/streamlink/stream/hls.py#L404-L416

<br>

In addition, I'm providing the content of fetched M3U8 playlists:

- Master playlist:

```shell
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-STREAM-INF:BANDWIDTH=*,AVERAGE-BANDWIDTH=*,RESOLUTION=426x240,FRAME-RATE=14.985,CODECS="avc1.42C01E,mp4a.40.2"
index_5.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=*,AVERAGE-BANDWIDTH=*,RESOLUTION=640x360,FRAME-RATE=29.970,CODECS="avc1.4D401E,mp4a.40.2"
index_1.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=*,AVERAGE-BANDWIDTH=*,RESOLUTION=854x480,FRAME-RATE=29.970,CODECS="avc1.640029,mp4a.40.2"
index_2.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=*,AVERAGE-BANDWIDTH=*,RESOLUTION=1280x720,FRAME-RATE=29.970,CODECS="avc1.640029,mp4a.40.2"
index_4.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=*,AVERAGE-BANDWIDTH=*,RESOLUTION=1920x1080,FRAME-RATE=29.970,CODECS="avc1.640029,mp4a.40.2"
index_3.m3u8

```

- The 1920x1080-variant playlist:

```shell
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:6
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-DISCONTINUITY-SEQUENCE:4
#EXT-X-PROGRAM-DATE-TIME:2012-12-21T00:00:00.000Z
#EXTINF:6.006,
index_3_0.ts?m=1329782400
#EXTINF:6.006,
index_3_1.ts?m=1329782400
#EXTINF:6.006,
index_3_2.ts?m=1329782400
#EXTINF:6.006,
index_3_3.ts?m=1329782400
#EXTINF:6.006,
index_3_4.ts?m=1329782400
#EXTINF:6.006,
index_3_5.ts?m=1329782400
#EXTINF:6.006,
index_3_6.ts?m=1329782400
#EXTINF:6.006,
index_3_7.ts?m=1329782400
#EXTINF:6.006,
index_3_8.ts?m=1329782400
#EXTINF:6.006,
index_3_9.ts?m=1329782400

```
